## Analýza zamestnancov VŠ

In [1]:
import pandas as pd

from pandasql import sqldf
# from sklearn import datasets

In [2]:
df = pd.read_csv('../data/preprocessed/regzam.csv', index_col=0)
display(df)

Employee_ID GivenName  FamilyName DegreePedagogicalAcademic  \
0      18799121690    Štefan       Kočan                      Doc.   
1      14002860472     Dušan       Korgo                       DOC   
2      17922733804       Ján     Buzalka                     prof.   
3      16290288493     Jozef   Stieranka                     prof.   
4      21401303068     Ľuboš       Wäldl                      doc.   
...            ...       ...         ...                       ...   
32725  12110964830      Joža      Spurný                      doc.   
32726  13500390262   Marcela   Harčárová                       NaN   
32727  17108478442      Jana    Juhásová                       NaN   
32728  15917135755   Miriama    Kubušová                       NaN   
32729  51680829014      Jana  Ochodnícka                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University   Employment_Type  \
0      Vysoká škola bezpečnostného manažérstva v Koši...            Docent   
1                                                    NaN               NaN   
2                                                    NaN               NaN   
3                             Akadémia Policajného zboru          Profesor   
4                             Akadémia Policajného zboru          Profesor   
...                                                  ...               ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...            Docent   
32726  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32727  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32728  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32729  Vysoká škola zdravotníctva a sociálnej práce s...  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
1                                                    NaN                  NaN   
2                                                    NaN                  NaN   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_Ratio                             Guarantee_StudyProgram  \
0                  100                                                NaN   
1                  NaN                                                NaN   
2                  NaN                                                NaN   
3                  100  ['bezpečnostnoprávna ochrana osôb a majetku', ...   
4                  100                                                NaN   
...                ...                                                ...   
32725              100                                                NaN   
32726               50                                                NaN   
32727               50                                                NaN   
32728               50                                           

In [4]:
id_set = set(df["Employee_ID"])
print(len(id_set))

32730


-> máme Dataset s 32730 riadkami, pričom žiadne 2 Employee_ID nie sú rovnaké

In [5]:
name_set = set(df["FamilyName"] + " " + df["GivenName"])
len(name_set)

31639

-> 31639 rôznych mien; zostáva zistiť, či pri duplikátnych menách sa jedná o tú istú osobu, alebo nie.
Tiež by nás mohlo zaujímať, ako ukladali osoby, ktorým sa zmenilo meno...

In [6]:
def find_name_duplicates(df):
    name_set = set()
    name_duplicates = set()
    duplicates_ids = set()
    for index, row in df.iterrows():
        name = row["FamilyName"] + " " + row["GivenName"]
        if name in name_set:
            name_duplicates.add(name)
        else:
            name_set.add(name)
    for index, row in df.iterrows():
        name = row["FamilyName"] + " " + row["GivenName"]
        if name in name_duplicates:
            duplicates_ids.add(row["Employee_ID"])
    return tuple(name_duplicates), tuple(duplicates_ids)

In [7]:
name_duplicates, ids = find_name_duplicates(df)

In [8]:
q = f"SELECT * FROM df WHERE Employee_ID IN {ids}"
duplicates_df = sqldf(q, globals()).sort_values(by=['FamilyName', 'GivenName'])
display(duplicates_df)

Employee_ID FamilyName  GivenName DegreePedagogicalAcademic  \
1327  49910822214  Adamíková   Katarína                      None   
1576  18913486096  Adamíková   Katarína                      None   
114   39501978150    Adamčík  František                     prof.   
1325  17240205086    Adamčík  František                      None   
133   51642863772   Antalová       Jana                      None   
...           ...        ...        ...                       ...   
1876  15552653387      Žilka      Tomáš                      None   
1107  10056354153    Žilková   Viktória                      None   
1994  14058627798    Žilková   Viktória                      None   
530   14639882132    Žuffová       Anna                      None   
1296  96699806419    Žuffová       Anna                      None   

     DegreeMaster DegreeDoctor  \
1327         Mgr.         None   
1576         Mgr.         None   
114          Ing.         CSc.   
1325         Ing.         PhD.   
133         MUDr.         PhD.   
...           ...          ...   
1876        MUDr.         None   
1107         Mgr.         None   
1994         None         MSc.   
530          None         None   
1296         Mgr.         None   

                     Employment_Workplace_University  \
1327                                            None   
1576                                            None   
114                 Technická univerzita v Košiciach   
1325                                            None   
133               Univerzita Komenského v Bratislave   
...                                              ...   
1876  Slovenská zdravotnícka univerzita v Bratislave   
1107                                            None   
1994              Univerzita Komenského v Bratislave   
530                                             None   
1296                                            None   

     Employment_Workplace_Faculty  \
1327                         None   
1576                         None   
114               Letecká fakulta   
1325                         None   
133              Lekárska fakulta   
...                           ...   
1876             Lekárska fakulta   
1107                         None   
1994             Lekárska fakulta   
530                          None   
1296                         None   

                                        Employment_Type  \
1327                                               None   
1576                                               None   
114                                            Profesor   
1325                                               None   
133                                    Odborný asistent   
...                                                 ...   
1876  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...   
1107                                               None   
1994                     Výskumný pracovník - výskumník   
530                                                None   
1296                                               None   

                                     Employment_Subtype Employment_BeginDate  \
1327                                               None                 None   
1576                                               None                 None   
114   s titulom profesor a vysokoškolským vzdelaním ...           2005-02-01   
1325                                               None                 None   
133          s vysokoškolským vzdelaním tretieho stupňa           2020-08-15   
...                                                 ...                  ...   
1876          s vysokoškolským vzdelaním druhého stupňa           2022-02-01   
1107                                               None                 None   
1994          s vysokoškolským vzdelaním druhého stupňa           2022-06-15   
530                                                None                 None   
1296                                               None      

In [9]:
df.query("GivenName == 'Viktória' and FamilyName == 'Žilková'")

Employee_ID FamilyName GivenName DegreePedagogicalAcademic  \
17361  10056354153    Žilková  Viktória                       NaN   
32149  14058627798    Žilková  Viktória                       NaN   

      DegreeMaster DegreeDoctor     Employment_Workplace_University  \
17361         Mgr.          NaN                                 NaN   
32149          NaN         MSc.  Univerzita Komenského v Bratislave   

      Employment_Workplace_Faculty                 Employment_Type  \
17361                          NaN                             NaN   
32149             Lekárska fakulta  Výskumný pracovník - výskumník   

                              Employment_Subtype Employment_BeginDate  \
17361                                        NaN                  NaN   
32149  s vysokoškolským vzdelaním druhého stupňa           2022-06-15   

      Employment_EndDate Employment_Ratio  Guarantee Guarantee_StudyProgram  \
17361                NaN              NaN      False                    NaN   
32149                NaN               50      False                    NaN   

      Guarantee_University Guarantee_Faculty Guarantee_FormOfStudy  \
17361                  NaN               NaN                   NaN   
32149                  NaN               NaN                   NaN   

      Guarantee_LevelOfStudy  
17361                    NaN  
32149                    NaN

... V tomto bode sme zistili, že naše stiahnuté xml zrejme obsahuje len dáta pre aktuálnych zamestnancov...

Pre minulých (zamestnancov) neobsahuje údaje o zamestnaní, predovšetkým nevieme z dát zistiť, v akom období a kde boli zamestnaní, hoci ich meno je v dátach.

In [23]:
df.query("Employment_EndDate.isna()")

Employee_ID  FamilyName GivenName DegreePedagogicalAcademic  \
0      18799121690       Kočan    Štefan                      Doc.   
1      14002860472       Korgo     Dušan                       DOC   
2      17922733804     Buzalka       Ján                     prof.   
3      16290288493   Stieranka     Jozef                     prof.   
4      21401303068       Wäldl     Ľuboš                      doc.   
...            ...         ...       ...                       ...   
32725  12110964830      Spurný      Joža                      doc.   
32726  13500390262   Harčárová   Marcela                       NaN   
32727  17108478442    Juhásová      Jana                       NaN   
32728  15917135755    Kubušová   Miriama                       NaN   
32729  51680829014  Ochodnícka      Jana                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University  \
0      Vysoká škola bezpečnostného manažérstva v Koši...   
1                                                    NaN   
2                                                    NaN   
3                             Akadémia Policajného zboru   
4                             Akadémia Policajného zboru   
...                                                  ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...   
32726  Vysoká škola zdravotníctva a sociálnej práce s...   
32727  Vysoká škola zdravotníctva a sociálnej práce s...   
32728  Vysoká škola zdravotníctva a sociálnej práce s...   
32729  Vysoká škola zdravotníctva a sociálnej práce s...   

      Employment_Workplace_Faculty   Employment_Type  \
0                              NaN            Docent   
1                              NaN               NaN   
2                              NaN               NaN   
3                              NaN          Profesor   
4                              NaN          Profesor   
...                            ...               ...   
32725                          NaN            Docent   
32726                          NaN          Asistent   
32727                          NaN          Asistent   
32728                          NaN          Asistent   
32729                          NaN  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
1                                                    NaN                  NaN   
2                                                    NaN                  NaN   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_EndDate Employment_Ratio  Guarantee  \
0                    NaN              100      False   
1                    NaN              NaN      False   
2                    NaN              NaN      False   
3                    NaN              100       True   
4                    NaN              100      False   
...                  ...              ... 

-> Až 32434 riadkov nemá EndDate, čo aj potvrdzuje našu hypotézu, že máme zlé dáta :(